### PR TITLE
Feature/connected grid

### PR DESCRIPTION
**Pull request by elvisduru**

### CODE REVIEW

File: src/components/Column/utils.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    }
+    return styles;
+};
+
+type CustomProps = {
+    [key: string]: ViewStyle;
+};
+
+export const columnStyles: ComponentStylePropWithVariants<ViewStyle, '', CustomProps> = {
+    ...generateColStyles(12),
```
Comment (shashank1010) [inline src/components/Column/utils.tsx:20]: if it's already returning a new object, I don't see a point of spreading it. It's a pattern I never understood

File: stories/components/Grid/Grid.docs.tsx [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+            </Text>
+        </View>
+    );
+};
+
+const firstCodeBlock = `
+import { memo, useCallback } from 'react';
+import type { ViewProps } from 'react-native';
+import Grid from '../../../src/components/Grid/Grid';
+import { useMolecules } from '../../../src/hooks';
```
Comment (shashank1010) [inline stories/components/Grid/Grid.docs.tsx:56]: import from "@bambooapp/bamboo-molecules"

Comment (shashank1010) [reply stories/components/Grid/Grid.docs.tsx:56]: Lets make it a working Demo.

All the components you require, should be created in the file itself.
By that I mean, do not import it from Grid, for the documentation only

File: stories/components/Grid/Grid.docs.tsx [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+            <View>
+                <Text>{item.title}</Text>
+                <Text>{item.value}</Text>
+            </View>
+        ),
+        [],
+    );
+    return (
+        <Card style={style}>
+            <FlatList
```
Comment (shashank1010) [inline stories/components/Grid/Grid.docs.tsx:114]: So, the demo works with FlashList?

File: src/components/Column/Column.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+const Column: FC<PropsWithChildren<Props>> = ({ breakPoints, children, alignment, style }) => {
+    const { View } = useMolecules();
+    const { width } = useWindowDimensions();
+    const colSize = useMemo(() => {
+        if (breakPoints === undefined) return 1;
+        if (typeof breakPoints === 'number') {
+            return breakPoints;
+        } else {
+            if (breakPoints.xl && width > breakpoints.lg) {
```
Comment (shashank1010) [inline src/components/Column/Column.tsx:49]: breakpoints, lets rename that to referenceBreakPoints

Comment (shashank1010) [reply src/components/Column/Column.tsx:49]: referenceBreakPoints come down from the grid as a prop

File: src/components/Grid/Grid.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            gridStyles: { ...grid, ...otherStyles },
+            rowStyles: row,
+        };
+    }, [componentStyles, width]);
+
+    const MemoizedRenderer = useRef(renderer).current;
+
+    const renderItem = useCallback(
+        ({ item }: { item: ColumnProps }) => (
+            <Column {...item}>
```
Comment (shashank1010) [inline src/components/Grid/Grid.tsx:32]: add referenceBreakPoints={breakpoints}

File: src/components/Grid/Grid.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+};
+
+const keyExtractor = (item: ColumnProps) => item.name;
+
+function Grid({ data, style, renderer }: Props, ref: any) {
+    const { View } = useMolecules();
+    const { width } = useWindowDimensions();
+    const componentStyles = useComponentStyles('Grid', style);
+
+    const { gridStyles, rowStyles } = useMemo(() => {
```
Comment (shashank1010) [inline src/components/Grid/Grid.tsx:19]: accept an optional breakpoint prop, if it is unavailable, take the breakpoints from the theme

**Review by shashank1010 — CHANGES_REQUESTED** [review]

File: src/components/Grid/Grid.tsx [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+    const MemoizedRenderer = useRef(renderer).current;
+
+    const renderItem = useCallback(
+        ({ item }: { item: ColumnProps }) => (
+            <Column {...item}>
+                <MemoizedRenderer name={item.name} />
+            </Column>
+        ),
+        [renderer],
```
Comment (thetaungg) [inline src/components/Grid/Grid.tsx:36]: remove renderer as a dependency

File: src/components/Grid/Grid.tsx [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    );
+
+    return (
+        <View ref={ref} style={gridStyles}>
```
Comment (thetaungg) [inline src/components/Grid/Grid.tsx:40]: View is not relevant, we can directly use Flatlist.

File: src/components/Grid/Grid.tsx [hunk 9]
```diff
@@ -0,0 +1,52 @@
+import { FC, forwardRef, memo, useCallback, useMemo, useRef } from 'react';
+import { FlatList, useWindowDimensions, ViewProps } from 'react-native';
+import { useComponentStyles, useMolecules } from '../../hooks';
+import type { ColumnProps } from '../Column';
+import Column from '../Column/Column';
+
+export type Props = ViewProps & {
```
Comment (thetaungg) [inline src/components/Grid/Grid.tsx:7]: Should be Flatlist props and pass down the extra props

**Review by thetaungg — CHANGES_REQUESTED** [review]

File: stories/components/Grid/Grid.docs.tsx (hunk 3 above)
Comment (elvisduru) [reply stories/components/Grid/Grid.docs.tsx:114]: Yeah, I tried using FlashList, but according to their docs currently they do not implement the `columnWrapperStyle` prop. And without that prop, I can't make the columns wrap using CSS. Sure, there may be a workaround to that, but we can always investigate that later.

Comment (elvisduru) [reply stories/components/Grid/Grid.docs.tsx:114]: Okay, for the Demo's table component, yes I can use a FlashList instead.

**Review by shashank1010 — DISMISSED** [review]

**Review by thetaungg — APPROVED** [review]

**Pull request merged**